CorsixTH/Src/th_sound.cpp

```diff
@@ -150,179 +150,179 @@ size_t THSoundArchive::getSoundDuration(size_t iIndex)
 
 #undef FOURCC
 
-SDL_RWops* THSoundArchive::loadSound(size_t iIndex)
+SDL_RWops* th_sound_archive::load_sound(size_t iIndex)
 {
-    if(iIndex >= m_iFileCount)
+    if(iIndex >= sound_file_count)
         return nullptr;
 
-    th_fileinfo_t *pFile = m_pFiles + iIndex;
-    return SDL_RWFromConstMem(m_pData + pFile->position, pFile->length);
+    th_fileinfo *pFile = sound_files + iIndex;
+    return SDL_RWFromConstMem(data + pFile->position, pFile->length);
 }
 
 #ifdef CORSIX_TH_USE_SDL_MIXER
 
-THSoundEffects* THSoundEffects::ms_pSingleton = nullptr;
+sound_player* sound_player::singleton = nullptr;
 
-THSoundEffects::THSoundEffects()
+sound_player::sound_player()
 {
-    m_ppSounds = nullptr;
-    m_iSoundCount = 0;
-    ms_pSingleton = this;
-    m_iCameraX = 0;
-    m_iCameraY = 0;
-    m_fCameraRadius = 1.0;
-    m_fMasterVolume = 1.0;
-    m_fSoundEffectsVolume = 0.5;
-    m_iPostionlessVolume = MIX_MAX_VOLUME;
-    m_bSoundEffectsOn = true;
+    sounds = nullptr;
+    sound_count = 0;
```

Comment:
why does a sound player need to know how many sounds exist

Reply:
Because this sound player prepares an SDL sound for every piece of sound data so that they are ready to play, and because they don't use an stl container for sounds.